framework/sandstone.cpp

```diff
@@ -1038,6 +1039,43 @@ int num_packages() {
     return Topology::topology().packages.size();
 }
 
+int num_enabled_cpus() {
+    return sApp->enabled_cpus_count;
+}
+
+void reschedule() {
+    // Choose a random CPU to hop to
+    int cpu_idx = random32() % num_enabled_cpus();
```

Comment:
We need to make this a command-line option. For example, `--cpu-reschedule=(none|random|queue)`. This way a test would have only one function to call (`reschedule()`) and the actual rescheduling behavior could be controlled from the command-line.

Reply:
`queue` should probably be the default setting (as it eliminates the chance of a CPU executing more than 1 opendcdiag thread at the same time).